research-software-directory/RSD-as-a-service
# fix: improve publication search by title for Crossref

# Mentions, impact and output search by publication title

Closes #512 

Changes proposed in this pull request:
*  Adding the mentions, impact and output publication items is improved. The crossref request are made faster and in case of failure of one of the calls the other results are shown.  
*  If user continues typing the searchterm while api request is in progress, we ignore previous request and use the latest input to perform another request. And we show only the results of the latest request
* Crossref api preformance is improved by sending the "polite" parameter witch contains our contact email. This information is stored in CROSSREF_CONTACT_EMAIL env variable

How to test:
* `docker-compose build`: to build frontend
*  Ensure you have CROSSREF_CONTACT_EMAIL in your env file with valid email
* `docker-compose up`: to start 
* create software and project if needed
* navigate to edit software and add mentions. Adding by DOI should work as it was in the past. Searching by title should be improved: it should take less time (still about 10sec) and in case of Crossref not responding within 30sec. it should return Datacite and RSD results.
* navigate to edit project and add input and output. The similar results should be achieved as with mentions on edit software page.

PR Checklist:
